File: src/facility/restaurant/Restaurant.java
/**
 * @author NIM / Nama : 13515087 / Audry Nyonata
 *  Nama file : Restaurant.java
 */

package facility.restaurant;

import java.util.*;

import facility.Facility;

import java.lang.*;

public class Restaurant extends Facility{
   /**
   * Constructor
   */
  public Restaurant(){
    super();
  }
  /**
   * Mengembalikan karakter untuk ditampilkan ke layar
   * @return id bertipe char
   */
  public char render(){
    return 'R';
  }
  /**
   * Mengembalikan warna untuk ditampilkan ke layar
   * @return color warna renderable
   */
  public String renderWithColor(){
    return ANSI_WHITE + render() + ANSI_RESET;
  }
}

File: virtual-zoo-cpp/vz03/src/renders/facilities/restaurant/restaurant.cpp
/* NIM/Nama  : 13515AAA / AAA AAA   */
/* File      : Restaurant.cpp       */

#include <iostream>
#include "restaurant.h"
using namespace std;

/***** RESTAURANT *****/
Restaurant::Restaurant() : Facility('R', WHITE)
{}

Restaurant::Restaurant(const Restaurant& R) : Facility(R)
{}

Restaurant::~Restaurant()
{}

Restaurant& Restaurant::operator=(const Restaurant& R)
{
  Facility::operator=(R);
  return *this;
}

char Restaurant::render()
{
	return id;
}

Color Restaurant::getColor()
{
	return color;
}

Restaurant* Restaurant::clone() const {
  return new Restaurant(*this);
}
